HealthVault/Classes/Caching/MHVThingCacheDatabaseProtocol.h

```diff
@@ -0,0 +1,156 @@
+//
+//  MHVThingCacheDatabaseProtocol.h
+//  MHVLib
+//
+// Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import <CoreData/CoreData.h>
+@class MHVThingCollection;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MHVCachedRecord
+
+// id<MHVCachedRecord> are treated as opaque objects by MHVThingCache.
+// MHVThingCacheDatabaseProtocol methods perform all Record actions needed by the MHVThingCache
+
+@end
+
+@protocol MHVThingCacheDatabaseProtocol <NSObject>
+
+/**
+ Create a new cache record given a recordId
+
+ @param recordId The record ID
+ @return the cache record
+ */
+- (id<MHVCachedRecord>)newRecord:(NSString *)recordId;
```

Comment:
We spoke offline about changing this to just using the recordId to provide more flexibility for consumers.